app/models/form/sales/subsections/shared_ownership_staircasing_transaction.rb

```diff
@@ -0,0 +1,35 @@
+class Form::Sales::Subsections::SharedOwnershipStaircasingTransaction < ::Form::Subsection
+  def initialize(id, hsh, section)
+    super
+    @id = "shared_ownership_staircasing_transaction"
+    @label = "Shared ownership - staircasing transaction"
+    @depends_on = [{ "ownershipsch" => 1, "setup_completed?" => true, "staircase" => 1 }]
+  end
+
+  def pages
+    @pages ||= [
+      (Form::Sales::Pages::Staircase.new(nil, nil, self) unless form.start_year_2025_or_later?),
+      Form::Sales::Pages::AboutStaircase.new("about_staircasing_joint_purchase", nil, self, joint_purchase: true),
+      Form::Sales::Pages::AboutStaircase.new("about_staircasing_not_joint_purchase", nil, self, joint_purchase: false),
+      Form::Sales::Pages::StaircaseSale.new(nil, nil, self),
+      Form::Sales::Pages::StaircaseBoughtValueCheck.new(nil, nil, self),
+      Form::Sales::Pages::StaircaseOwnedValueCheck.new("staircase_owned_value_check_joint_purchase", nil, self, joint_purchase: true),
+      Form::Sales::Pages::StaircaseOwnedValueCheck.new("staircase_owned_value_check_not_joint_purchase", nil, self, joint_purchase: false),
+      Form::Sales::Pages::StaircaseFirstTime.new(nil, nil, self),
+      Form::Sales::Pages::StaircasePrevious.new(nil, nil, self),
+      Form::Sales::Pages::StaircaseInitialDate.new(nil, nil, self),
+      Form::Sales::Pages::ValueSharedOwnership.new("value_shared_ownership_staircasing", nil, self),
+      Form::Sales::Pages::AboutPriceValueCheck.new("about_price_shared_ownership_value_check", nil, self),
+      Form::Sales::Pages::Equity.new(nil, nil, self),
+      Form::Sales::Pages::SharedOwnershipDepositValueCheck.new("shared_ownership_equity_value_check", nil, self),
+      Form::Sales::Pages::Mortgageused.new("mortgage_used_shared_ownership", nil, self, ownershipsch: 1),
+      Form::Sales::Pages::MonthlyRentStaircasingOwned.new(nil, nil, self),
+      Form::Sales::Pages::MonthlyRentStaircasing.new(nil, nil, self),
+      Form::Sales::Pages::MonthlyChargesValueCheck.new("monthly_charges_shared_ownership_value_check", nil, self),
+    ].compact
+  end
+
+  def displayed_in_tasklist?(log)
+    (log.ownershipsch.nil? || log.ownershipsch == 1) && (log.staircase.nil? || log.staircase == 1)
```

Comment:
Do we want to display it if the ownership and staircase is not answered?

Reply:
I was thinking we'd show the section until it gets ruled out. But we could hide the section until it becomes valid, i'll change it.

Reply:
Updated to match what you've done in the other section